bootstrap/kubeadm/main.go

```diff
@@ -75,6 +78,15 @@ func InitFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.")
 
+	fs.DurationVar(&leaseDuration, "lease-duration", 15*time.Second,
+		"Interval at which non-leader candidates will wait to force acquire leadership (e.g. 20s)")
+
+	fs.DurationVar(&renewDeadline, "renew-deadline", 10*time.Second,
+		"Duration that the acting master will retry refreshing leadership before giving up (e.g. 15s)")
+
+	fs.DurationVar(&retryPeriod, "retry-period", 2*time.Second,
+		"Duration the LeaderElector clients should wait between tries of actions (e.g. 5s)")
```

Comment:
Should we add some validation to ensure that the values are sane?

Probably an issue for followup, but we are replicating a lot of the same flags and logic between the different controllers (and also expecting providers to implement similar), I'm wondering if we need to evaluate a library for providing the common functionality.

Reply:
Yeah that would make more sense than having every controller implement its own subset of options. You're right that this would make for another issue with a broader scope.

Reply:
Regarding validation and sanity, do you mean parsing the values to ensure they meet the type (i.e. using `time.ParseDuration()` to validate) or do you mean making sure those values make sense (i.e. setting upper and lower limits)? Or both?

In any case, if we do that for the newly added values I see no reason why we shouldn't do it with the rest of values too. Open to anything, just lmk.

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/2710